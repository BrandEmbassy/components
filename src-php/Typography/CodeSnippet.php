<?php declare(strict_types = 1);

namespace BrandEmbassy\Components\Typography;

use BrandEmbassy\Components\UiComponent;
use function htmlspecialchars;

final class CodeSnippet implements UiComponent
{
    /**
     * @var string
     */
    private $code;


    public function __construct(string $code)
    {
        $this->code = $code;
    }


    public function render(): string
    {
        return '
        <pre style="
                white-space: pre-wrap;
                box-sizing: border-box;
                background-color: #f4f4f4;
                width: 100%;
                margin-top: 20px;
                padding: 14px;
            ">
            ' . htmlspecialchars($this->code) . '
        </pre>
        ';
    }
}
