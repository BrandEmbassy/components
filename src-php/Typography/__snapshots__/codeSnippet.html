<pre style="
    white-space: pre-wrap;
    box-sizing: border-box;
    background-color: #f4f4f4;
    width: 100%;
    margin-top: 20px;
    padding: 14px;">
    &lt;script type=&quot;text/javascript&quot;&gt;alert('Hello boys &amp; girls!')&lt;/script&gt;
</pre>
